backend/tests/unit/services/test_agent_registry.py

```
@@ -402,4 +341,278 @@
+        assert "Existing agent: 'AgentX'" in warning_msg
+        assert "New agent: 'AgentY'" in warning_msg
+        assert "from chain 'chain2'" in warning_msg
+        assert "Keeping existing mapping" in warning_msg
+    
+    @patch("tarsy.services.agent_registry.BUILTIN_CHAIN_DEFINITIONS")
+    def test_extract_mappings_with_consistent_duplicates(self, mock_chain_defs, mock_chain_definitions_with_duplicates, caplog):
+        """Test that duplicate mappings with same agent are logged as debug."""
+        mock_chain_defs.items.return_value = mock_chain_definitions_with_duplicates.items()
+        
+        with caplog.at_level("DEBUG"):
+            mappings = _extract_alert_type_mappings_from_chains()
```

Ruff (0.12.2)

Local variable `mappings` is assigned to but never used

Remove assignment to unused variable `mappings`

(F841)